### DISCUSSION TITLE
The viewer gets confused about inside vs. outside

### DISCUSSION THREAD

**Discussion by tve** [sgenoud/replicad#92] [Q&A] [answered]
In the [swept profile box example](https://replicad.xyz/docs/recipes/sweeped-profile-box) the viewer is confused about what's inside and what's outside if I plot the `side` in `baseSketch`:
![image](https://github.com/sgenoud/replicad/assets/39480/9259b923-6065-42bd-ae52-859df4392411)
Given that the profile is not closed this is understandable. How does the viewer decide what is inside vs. what is outside and how can it be influenced/fixed?

**Comment by sgenoud** [comment, marked as answer]
The issue is not with the viewer, but with the orientation of the shape itself (which depends on the direction you create them). The viewer recieves the orientation of the shape and display it as such.

replicad does not expose the method that would help you here, but you can access it from within the opencascade API. (I should add higher level API to do this.

`side.wrapped.Reverse()` does the trick. You can see it [here](https://studio.replicad.xyz/workbench?code=UEsDBAoAAAAIAOBa%252FlZfswYUGAIAAGwEAAAHAAAAY29kZS5qc4VTUWvbMBB%252B9684%252BiSD6yYpfVnIYCvt02AhKXQj9EGWLo6oIhlJbjpK%252FvtOsuO4XWGB2Dr5vu9On74T1vgAb7Dnz7i2WskiLe%252B5wAK497ivND4qR9GdrPFeGYmugBoNOiXWB8SmgKXmBuEIiwzAYaOV4HKeZdvWiKCsgcbZrdL43b4yZZo2LLu4gIp7zOGNYCL14QN3ARYwzior3WLjlAnlVjkflpaW8wHSc38EBceN1zwgu0ykm8lTAf1y%252BpRTdycCNJLAzT%252FFNB9qDbmx3%252FUzBrEjSAxKn6KfJknAOt6rK3hEEA6pPIQdglcSwW7TurKv5%252BPG%252FcWItRTaRprSR2G7PUbpAKyJBXJYfE1yxZ%252FD0LpB3A%252BddOnzlHos0uuEO6iwu7UmcBG%252BQHAtdp%252BP9Iz90yv2VR4cbxqU5Qpf0HlqKhtKxu%252Fng5LqwLVOp2sUCvQQbI0UujNk8BfbpGqRoqtLJA%252BE3JLjOluRahKqP5A0UKZOxPGWegX7AyfwyatsbFVm8DByK8mpelUufv2%252BKCJX9ADZyUgWG8nz%252FD%252B9pJuja0pZo%252FuKmSyBo6eOWSb6eZKOH4r0XNmWupArFIGbWqc5%252BXRK9lwZNp6GWIdyPyNh15MCZvS%252FGVv5PAsRk24MoNSKTn4TU0fxrIDrPt79iBuXsz586cJpH1atrvGbEw%252BWbajelMZoUp5ypR1M%252F96NcdSbd1NO6vwFUEsBAhQACgAAAAgA4Fr%252BVl%252BzBhQYAgAAbAQAAAcAAAAAAAAAAAAAAAAAAAAAAGNvZGUuanNQSwUGAAAAAAEAAQA1AAAAPQIAAAAA)
